experiments/bert/bert.py

```diff
@@ -0,0 +1,231 @@
+import math
+
+import numpy as np
+import pandas as pd
+import torch
+from pytorch_pretrained_bert import BertTokenizer, BertForSequenceClassification
+from pytorch_pretrained_bert.optimization import BertAdam
+from torch.nn.utils import clip_grad_norm_
+from torch.optim import Optimizer
+from torch.optim.optimizer import required
+from tqdm import tqdm
+
+from transfer_nlp.loaders.loaders import DatasetSplits, DatasetHyperParams, DataFrameDataset
+from transfer_nlp.loaders.vectorizers import Vectorizer
+from transfer_nlp.loaders.vocabulary import Vocabulary
+from transfer_nlp.plugins.config import register_plugin
+
+tqdm.pandas()
+
+
+@register_plugin
+class BertVectorizer(Vectorizer):
+    def __init__(self, data_file: str):
+        super().__init__(data_file=data_file)
+        self.tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+
+        df = pd.read_csv(data_file)
+        target_vocab = Vocabulary(add_unk=False)
+        for category in sorted(set(df.category)):
+            target_vocab.add_token(category)
+        self.target_vocab = target_vocab
+
+    def vectorize(self, title: str, max_seq_length: int) -> np.array:
```

Comment:
return value type should be a Tuple[array * 3] right?

Reply:
right